File: com/problem/java/googleKickstart/2022/roundg/walktober3.java
import java.io.PrintStream;
import java.util.*;
import java.util.stream.IntStream;

public class Solution {
  static final PrintStream out = System.out;
  static Scanner sc = new Scanner(System.in);
  static void solve() {
    int N = sc.nextInt();
    int M = sc.nextInt();
    int P = sc.nextInt();
    int ans = 0;

    HashMap<Integer, ArrayList<Integer>> scoreBoard = new HashMap<>();
    for (int i = 1; i <= N; ++i) 
    {
      ArrayList<Integer> score = new ArrayList<>();
      for (int j = 1; j <= M; ++j) 
      {
        int S = sc.nextInt();
        score.add(S);
      }
      scoreBoard.put(i, score);
    }

    ArrayList<Integer> john = new ArrayList<>();
    for (int i = 0; i < N; i++) 
    {
      john.add(scoreBoard.get(P).get(i));


    }

    out.println(ans);
  }

  public static void main(String[] args) {
    int T = sc.nextInt();

    for (int t = 1; t <= T; ++t) {
      //out.println("Case #"+t+": ");
      solve();
    }

  }
}

File: com/problem/go/googleKickstart/2022/roundg/walktober3.go
package main

import (
  "fmt"
)

func solve(N int, M int, id int) {
  /**
  * 1. put all scores in a double array [i][j]
  * 2. get all john scores by john id
  * 3. find max scores each day
  * 4. get the diff number for john to compete
  **/
  var scoreBoard = make([][]int, N)
  var maxScores = make([]int, M)
  var john = make([]int, N)
  var ans int

  for i := 1; i <= N; i++ {
    var score = make([]int, N)
    for j := 1; j <= M; j++ {
      var S int
      fmt.Scan(&S)
      score[j - 1] = S
      scoreBoard[i][j] = score[j]
    }
  }

  for i := 0; i < M; i++ {
    john[i] = scoreBoard[id][i]
  }

  for k := 0; k < M; k++ {
    var max int
    for i := 0; i < len(scoreBoard); i++ {
      if scoreBoard[i][k] > max {
        max = scoreBoard[i][k]
      }
    }
    maxScores[k] = max
  }

  for i := 0; i < M; i++ {
    var diff int
    if john[i] < maxScores[i] {
      diff = maxScores[i] - john[i];
    }
    if diff > 0 {
      ans += diff
    }
    // fmt.Println("diff : "+diff)
  }

  fmt.Println(ans)
}

func main() {
  var T int
  fmt.Scan(&T)
  for t := 1; t <= T; t++ {
    var ( N int; M int; id int )
    fmt.Scanf("%d %d %d", &N, &M, &id)
    fmt.Printf("Case #%d: ", t)
    solve(N, M, id)
  }
}
